buildscripts/templates/registry/incubating_java/IncubatingSemanticAttributes.java.j2

```diff
@@ -0,0 +1,108 @@
+{%- macro to_java_type(attribute) -%}
+  {%- if attribute.type == "string" -%}
+    String
+  {%- elif attribute.type == "string[]" -%}
+    List<String>
+  {%- elif attribute.type == "boolean" -%}
+    Boolean
+  {%- elif attribute.type == "int" -%}
+    Long
+  {%- elif attribute.type == "double" -%}
+    Double
+  {%- elif attribute.type is mapping -%}
+    {# Note: Treat all enums as strings for now? #}
+    String
+  {%- elif attribute.type == "template[string]" -%}
+    String
+  {%- elif attribute.type == "template[string[]]" -%}
+    List<String>
+  {%- else -%}
+    {{ attribute.type }}
+  {%- endif -%}
+{%- endmacro %}
+{%- macro to_java_key_factory(attribute) -%}
+  {%- if attribute.type == "string" -%}
+    stringKey
+  {%- elif attribute.type == "string[]" -%}
+    stringArrayKey
+  {%- elif attribute.type == "boolean" -%}
+    booleanKey
+  {%- elif attribute.type == "int" -%}
+    longKey
+  {%- elif attribute.type == "double" -%}
+    doubleKey
+  {%- elif attribute.type == "template[string]" -%}
+    stringTemplate
+  {%- elif attribute.type == "template[string[]]" -%}
+    stringArrayKeyTemplate
+  {%- elif attribute.type is mapping -%}
+    stringKey
+  {%- else -%}
+    {{ type }}Key
+  {%- endif -%}
+{%- endmacro %}
```

Comment:
We could generalize the `type_mapping` section in weaver.yaml and support multiple mapping filters.
See https://github.com/open-telemetry/weaver/issues/156

Reply:
I like this idea.

Reply:
I used this.  It still doesn't handle enums properly.